File: packages/engine/src/triggers/cron.ts
import type { EngineConfig } from "../config.js"

export function registerCronTriggers(sdk: any, config: EngineConfig) {
  sdk.registerTrigger({
    type: "cron",
    function_id: "lifecycle::ttl-sweep",
    config: { expression: config.ttlSweepInterval },
  })
}

File: packages/worker/src/triggers/cron.rs
use iii_sdk::III;
use serde_json::json;
use std::sync::Arc;

use crate::config::EngineConfig;

pub fn register(bridge: &Arc<III>, config: &EngineConfig) {
    let _ = bridge.register_trigger("cron", "lifecycle::ttl-sweep", json!({
        "expression": config.ttl_sweep_interval,
    }));
}
